packages/client/src/runtime/query.ts

```diff
@@ -1507,12 +1533,12 @@ function tryInferArgs(
 
   return new Arg({
     key,
-    value: value.map((v) => {
+    value: value.map((v, i) => {
       if (inputType.isList && typeof v !== 'object') {
         return v
       }
-      if (typeof v !== 'object' || !value) {
-        return getInvalidTypeArg(key, v, arg, inputType)
+      if (typeof v !== 'object' || !value || Array.isArray(v)) {
```

Comment:
This line is an actual fix, the rest is just ensuring that the error renders correctly.

Reply:
Maybe add some comments? Because it's hard to understand what's happening in general in `query.ts` 😅

Reply:
Added. Not sure how much they help in a grand scheme of things though.